### PR TITLE
Ace editor navigation

### PR DESCRIPTION
**Pull request by pujun-ai**
@percyliang @kashizui please review. 

This is for enabling https://github.com/codalab/codalab/pull/1399
check this out https://www.dropbox.com/s/qptwyexd125cefs/editor-navigation.mp4?dl=0

### CODE REVIEW

File: codalab/lib/worksheet_util.py [hunk 1]
```diff
@@ -571,8 +572,9 @@ def flush_bundles():
         elif mode == 'contents' or mode == 'image' or mode == 'html':
             def raiseUsageError():
                 raise UsageError('Expected \'% display ' + mode + ' (genpath)\', but got \'% display ' + ' '.join([mode] + args) + '\'')
-            for bundle_info in bundle_infos:
+            for i, bundle_info in bundle_infos:
```
Comment (kashizui) [inline codalab/lib/worksheet_util.py]: If this `i` has some meaning beside the number of iterations so far in the for loop, might be helpful to give it a more explicit name like `item_index`

File: codalab/lib/worksheet_util.py [hunk 2]
```diff
@@ -683,17 +689,20 @@ def get_command(value_obj):  # For directives only
...
                 # Combine all consecutive markup items
                 new_items[-1]['interpreted'] += '\n' + value_obj
+                raw_items_interpreted_items_map[i] = (len(new_items) - 1, 0)
             elif not new_last_was_empty_line:
                 new_items.append({
                     'mode': TYPE_MARKUP,
                     'interpreted': value_obj,
                     'properties': {},
                 })
+                raw_items_interpreted_items_map[i] = (len(new_items) -1, 0)
```
Comment (kashizui) [inline codalab/lib/worksheet_util.py]: nit: space after `-` sign

File: codalab/lib/worksheet_util.py [hunk 3]
```diff
@@ -541,6 +541,7 @@ def interpret_items(schemas, items):
     - ('search', [keyword, ...])
     '''
     result = {}
+    raw_items_interpreted_items_map = {}
```
Comment (percyliang) [inline codalab/lib/worksheet_util.py]: Comment precisely what are the keys and values of these two maps.

**Comment by percyliang** [comment]
LGTM modulo comments

File: codalab/lib/worksheet_util.py (hunk 1 above)
Comment (pujun-ai) [inline codalab/lib/worksheet_util.py]: made this change. Thanks :+1:

**Pull request merged**